jakarta-eclipse/src/org/jakarta/codeAction/ListenerImplementationQuickFix.java

```diff
@@ -0,0 +1,83 @@
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.jakarta.codeAction;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import org.jakarta.jdt.ServletConstants;
+import org.jakarta.codeActionProposal.ChangeCorrectionProposal;
+import org.jakarta.codeActionProposal.ExtendClassProposal;
+import org.jakarta.codeActionProposal.ImplementInterfaceProposal;
+
+/**
+ * QuickFix for fixing HttpServlet extension
+ * error by providing the code actions which implements
+ * IJavaCodeActionParticipant
+ *
+ * Adapted from https://github.com/eclipse/lsp4mp/blob/master/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/health/java/ImplementHealthCheckQuickFix.java
+ *
+ * @author Credit to Angelo ZERR
+ *
+ */
+public class ListenerImplementationQuickFix implements IJavaCodeActionParticipant{
+	@Override
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+			IProgressMonitor monitor) throws CoreException {
+		ASTNode node = context.getCoveredNode();
+		ITypeBinding parentType = Bindings.getBindingOfParentType(node);
+		if (parentType != null) {
+			List<CodeAction> codeActions = new ArrayList<>();
+			// Create code action
+			// interface
+			
+			CodeAction ca0 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.SERVLET_CONTEXT_LISTENER, "jakarta.servlet.ServletContextListener");
+			CodeAction ca1 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.SERVLET_CONTEXT_ATTRIBUTE_LISTENER, "jakarta.servlet.ServletContextAttributeListener");
+			CodeAction ca2 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.SERVLET_REQUEST_LISTENER, "jakarta.servlet.ServletRequestListener");
+			CodeAction ca3 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.SERVLET_REQUEST_ATTRIBUTE_LISTENER, "jakarta.servlet.ServletRequestAttributeListener");
+			CodeAction ca4 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.HTTP_SESSION_LISTENER, "jakarta.servlet.HttpSessionListener");
+			CodeAction ca5 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.HTTP_SESSION_ATTRIBUTE_LISTENER, "jakarta.servlet.HttpSessionAttributeListener");
+			CodeAction ca6 = setUpCodeAction(parentType, diagnostic, context, ServletConstants.HTTP_SESSION_ID_LISTENER, "jakarta.servlet.HttpSessionIdListener");
```

Comment:
should be:
jakarta.servlet.http.HttpSessionListener
jakarta.servlet.http.HttpSessionAttributeListener
jakarta.servlet.http.HttpSessionIdListener